src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1892,12 +1892,33 @@ static bool processSubscriptions
       //
       if (subCacheActive == false)
       {
-        BSONObj query  = BSON("_id" << OID(mapSubId));
-        BSONObj update = BSON("$set" <<
-                              BSON(CSUB_LASTNOTIFICATION << rightNow) <<
-                              "$inc" << BSON(CSUB_COUNT << (long long) 1));
+        BSONObj subOrig;
+        std::string err;
+        collectionFindOne(getSubscribeContextCollectionName(tenant), BSON("_id" << OID(mapSubId)), &subOrig, &err);
+        std::string status;
+        if (!subOrig.isEmpty())
+        {
+          if (subOrig.hasField(CSUB_STATUS))
+          {
+            status = getStringFieldF(subOrig, CSUB_STATUS);
+          }
+        }
 
-        ret = collectionUpdate(getSubscribeContextCollectionName(tenant), query, update, false, err);
+        BSONObj query  = BSON("_id" << OID(mapSubId));
+        BSONObj update;
+        // Update the value of status (in case of oneshot) in DB when broker is running without subscription cache
+        if (status == STATUS_ONESHOT)
+        {
+          update = BSON("$set" << BSON(CSUB_LASTNOTIFICATION << rightNow) <<
```

Comment:
It would be more correct to group both set statements. If I remember correctly it would be like:

          update = BSON("$set" << BSON(CSUB_LASTNOTIFICATION << rightNow << CSUB_STATUS << STATUS_INACTIVE) <<
                        "$inc" << BSON(CSUB_COUNT << (long long) 1));

Reply:
fixed in b2b21a1